ImageLounge/src/DkGui/DkNoMacs.cpp

```diff
@@ -2074,16 +2153,21 @@ DkNoMacsFrameless::DkNoMacsFrameless(QWidget *parent, Qt::WindowFlags flags)
     am.action(DkActionManager::menu_view_frameless)->setChecked(true);
     am.action(DkActionManager::menu_view_frameless)->blockSignals(false);
 
+    // frameless window simply goes on the same monitor as normal window
+    QScreen *screen = qApp->screenAt(geometry().topLeft());
+    setGeometry(screen->availableGeometry());
+
+    // TODO: if this is acceptable, remove menu_view_monitors, chooseMonitor(), DkMonitorConfigDialog, etc
+#ifdef CHOOSE_MONITOR
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     mDesktop = QApplication::desktop();
-
     chooseMonitor(false);
 #endif
-    show();
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     connect(mDesktop, &QDesktopWidget::workAreaResized, this, &DkNoMacsFrameless::chooseMonitor);
     connect(am.action(DkActionManager::menu_view_monitors), &QAction::triggered, this, &DkNoMacsFrameless::chooseMonitor);
+#endif
 #endif
```

Comment:
This would disable this block completely because CHOOSE_MONITOR is not defined anywhere. Is it necessary and safe to do so in this PR?

Reply:
It is safe to do, but not necessary. I am not aware of the reasoning around this design choice, but I think it may be a good thing to deprecate:

- It's pretty atypical behavior for a viewer application
- The frameless window might show on a different monitor from the normal window; one might not remember or be confused as to why that is.
- If you clicked "Remember my choice" in choose monitor dialog, you might have trouble changing monitors; I can't seem to find the context menu anywhere.

Reply:
I agree with the reasoning. However, if this deprecation is unnecessary for this fix to work, maybe we should split this change into another PR. That PR could remove this whole block instead of hiding part of the code by a macro, and unused functions like `chooseMonitor` could also be cleaned up.

Reply:
Agreed, I'll back out this change. I think that wraps things up unless you have anything else?

Reply:
After further reflection I've ported chooseMonitor to Qt6 to avoid any functional bifurcation. The new implementation works on both Qt versions.